### DISCUSSION TITLE
Question about using assume_straight_pages

### DISCUSSION THREAD

**Discussion by WalidHadri-Iron** [mindee/doctr#1611] [General]
Hello,

It's mentioned that using "assume_straight_pages: if True, speeds up the inference by assuming you only pass straight pages", my question is what does assume_straight_pages set as True make the inference behaves differently from using it when it's set as False. My only finding is that the difference is between using extract_crops (when True) extract_rcrops (When False), are the CV2 calculation in extract_rcrops the cause behind it. 

Thanks !

**Comment by felixdittrich92** [comment]
Hi @WalidHadri-Iron :wave:,

1. with `assume_straight_pages=False` we extract and work with 4-point polygons ((4, 2)) instead of boxes
2. we make use of an small (mobilenet_v3_small) crop orientation model to get the general orientation of each crop / with the next release we will extend this by predict also the general page orientation which enables us to handle also upside down cases (currently only in range -90 - 90 degree)
3. we estimate also the exact page orientation 

Best regards,
Felix

PS: what do you mean with "difference" ? The results should be more precise also if your doc contains only small rotations

**Reply by WalidHadri-Iron** [reply]
Hi @felixdittrich92,

What I meant by the difference is what are the things that happen when we set it as True, that wouldn't happen when it's False (and probably the other way around). But I think your points answered my question. Thanks a lot !
